src/main/java/com/commercetools/sync/products/utils/ProductVariantUpdateActionUtils.java

```diff
@@ -196,31 +199,27 @@
             final int variantId,
             @Nonnull final List<Image> oldImages,
             @Nonnull final List<Image> newImages) throws IllegalArgumentException {
-
-        if (oldImages.size() != newImages.size()) {
+        final int oldImageListSize = oldImages.size();
+        final int newImageListSize = newImages.size();
+        if (oldImageListSize != newImageListSize) {
             throw new IllegalArgumentException(
-                format("Old and new image lists must have the same size, but they have %d and %d respectively",
-                    oldImages.size(), newImages.size()));
+                format(IMAGE_LISTS_NOT_THE_SAME_SIZE, oldImageListSize, newImageListSize));
         }
 
-        final int listSize = oldImages.size();
-
         // optimization: to avoid multiple linear image index searching in the loop below - create an [image -> index]
         // map. This avoids quadratic order of growth of the implementation for large arrays.
-        final Map<Image, Integer> imageIndexMap = new HashMap<>(listSize);
+        final Map<Image, Integer> imageIndexMap = new HashMap<>(oldImageListSize);
         int index = 0;
         for (Image newImage : newImages) {
             imageIndexMap.put(newImage, index++);
         }
 
         final List<MoveImageToPosition> updateActions = new ArrayList<>();
 
-        for (int oldIndex = 0; oldIndex < listSize; oldIndex++) {
+        for (int oldIndex = 0; oldIndex < oldImageListSize; oldIndex++) {
             final Image oldImage = oldImages.get(oldIndex);
-
             final Integer newIndex = ofNullable(imageIndexMap.get(oldImage)) // constant-time operation
```

Comment:
@andrii-kovalenko-ct do you think it makes sense to leave such optimiztion comments? like here `// constant-time operation` and at line 209 `// optimization: to avoid multiple linear image index searching in the loop below - create an [image -> index]
          // map. This avoids quadratic order of growth of the implementation for large arrays.`

Reply:
if i put them there this definitely means _i think it makes sense to leave such comments_. This might be not true for other people though. 
But in cases i have some doubts _this might be understood wrong in the future_, i always leave such a comment: better to have redundancy rather than unclearness.

If you think it is too verbose - feel free to vanish it

Reply:
I just felt it was just left there as an explanation for the change from the previous implementation. Not as a general explanation. I guess it causes no harm to leave it maybe sometimes better to be clear than clean.

-- No need to "vanish" it :D